File: app/src/main/java/com/jacosro/justdraw/util/Paints.java
package com.jacosro.justdraw.util;

import android.graphics.Color;
import android.graphics.Paint;
import android.graphics.PorterDuff;
import android.graphics.PorterDuffXfermode;

import org.jetbrains.annotations.NotNull;

import java.util.Random;

public class Paints {

    public static final float DEFAULT_STROKE_WIDTH = 30f;
    public static final int DEFAULT_COLOR = Color.BLACK;

    private static Paint basePaint() {
        Paint paint = new Paint();
        paint.setAntiAlias(true);
        paint.setStrokeWidth(DEFAULT_STROKE_WIDTH);
        paint.setStyle(Paint.Style.STROKE);
        paint.setStrokeJoin(Paint.Join.ROUND);
        paint.setStrokeCap(Paint.Cap.ROUND);

        return paint;
    }

    public static Paint getRandomColorPaint() {
        Random random = new Random();
        Paint paint = basePaint();
        paint.setARGB(random.nextInt(), random.nextInt(), random.nextInt(), random.nextInt());

        return paint;
    }

    public static Paint getPaintWithColor(int color) {
        Paint paint = basePaint();
        paint.setColor(color);

        return paint;
    }

    public static Paint getErasePaint() {
        return getPaintWithColor(Color.WHITE);
    }


    @NotNull
    public static Paint getDefault() {
        return getPaintWithColor(DEFAULT_COLOR);
    }
}

File: app/src/main/java/com/jacosro/justdraw/util/Paints.kt
package com.jacosro.justdraw.util

import android.graphics.Color
import android.graphics.Paint
import android.graphics.PorterDuff
import android.graphics.PorterDuffXfermode

import java.util.Random

object Paints {

    const val DEFAULT_STROKE_WIDTH = 30f
    const val DEFAULT_COLOR = Color.BLACK

    val randomColorPaint: Paint
        get() {
            val random = Random()
            val paint = basePaint()
            paint.setARGB(random.nextInt(), random.nextInt(), random.nextInt(), random.nextInt())

            return paint
        }

    val erasePaint: Paint
        get() = getPaintWithColor(Color.WHITE)


    val default: Paint
        get() = getPaintWithColor(DEFAULT_COLOR)

    private fun basePaint(): Paint {
        val paint = Paint()
        paint.isAntiAlias = true
        paint.strokeWidth = DEFAULT_STROKE_WIDTH
        paint.style = Paint.Style.STROKE
        paint.strokeJoin = Paint.Join.ROUND
        paint.strokeCap = Paint.Cap.ROUND

        return paint
    }

    fun getPaintWithColor(color: Int): Paint {
        val paint = basePaint()
        paint.color = color

        return paint
    }
}